publication-rest/src/main/java/no/unit/nva/publication/download/CreatePresignedDownloadUrlHandler.java

```diff
@@ -0,0 +1,151 @@
+                                             UriShortener uriShortener,
+                                             HttpClient httpClient,
+                                             IdentityServiceClient identityServiceClient) {
+        super(Void.class, environment, httpClient);
+        this.resourceService = resourceService;
+        this.s3Presigner = s3Presigner;
+        this.uriShortener = uriShortener;
+        this.identityServiceClient = identityServiceClient;
+        this.basePath = environment.readEnv(CUSTOM_DOMAIN_BASE_PATH_ENV);
+    }
+
+    /**
+     * Default constructor for CreatePresignedDownloadUrlHandler.
+     */
+    @JacocoGenerated
+    public CreatePresignedDownloadUrlHandler() {
+        this(ResourceService.defaultService(),
+             defaultS3Presigner(),
+             new Environment(),
+             UriShortenerImpl.createDefault(),
+             HttpClient.newHttpClient(),
+             IdentityServiceClient.prepare());
+    }
+
+    @Override
+    protected void validateRequest(Void unused, RequestInfo requestInfo, Context context) throws ApiGatewayException {
+        //Do nothing
+    }
+
+    @Override
+    protected PresignedUriResponse processInput(Void input, RequestInfo requestInfo, Context context)
+        throws ApiGatewayException {
+        var publication = resourceService.getPublicationByIdentifier(RequestUtil.getIdentifier(requestInfo));
+        var file = getFileInformation(publication, requestInfo);
+
+        if (!hasFileAccess(publication, file, requestInfo)) {
+            throw new NotFoundException(
+                String.format(REQUESTED_RESOURCE_NOT_FOUND, publication.getIdentifier(), file.getIdentifier()));
+        }
+
+        return getPresignedUriResponse(file);
+    }
+
+    private PresignedUriResponse getPresignedUriResponse(File file) throws S3ServiceException {
+        var expiration = defaultExpiration();
+        var preSignedUriLong = getPresignedDownloadUrl(file, expiration);
+        var shortenedPresignUri = uriShortener.shorten(preSignedUriLong.id(), basePath, expiration);
+        return new PresignedUriResponse(file.getIdentifier().toString(), preSignedUriLong.id(), expiration, shortenedPresignUri);
+    }
+
+    private boolean hasFileAccess(Publication publication, File file, RequestInfo requestInfo) {
+        var userInstance = attempt(
+            () -> RequestUtil.createUserInstanceFromRequest(requestInfo, identityServiceClient)).or(() -> null).get();
+
+        var permissionStrategy = userInstance!= null ? PublicationPermissionStrategy.create(publication, userInstance,
+                                                                                            resourceService) : null;
+
+        return PublicationStatus.PUBLISHED.equals(publication.getStatus()) && file.isVisibleForNonOwner() ||
+               permissionStrategy != null && permissionStrategy.allowsAction(PublicationOperation.UPDATE);
+    }
```

Comment:
Too many nulls checking and assigning, maybe use optional?

Solution we discussed:

```
    private boolean hasFileAccess(Publication publication, File file, RequestInfo requestInfo) {
        var userInstance = attempt(
            () -> RequestUtil.createUserInstanceFromRequest(requestInfo, identityServiceClient)).toOptional();

        if (userInstance.isPresent()
            && PublicationPermissionStrategy.create(publication, userInstance.get(), resourceService).allowsAction(PublicationOperation.UPDATE)) {
            return true;
        }
        return PublicationStatus.PUBLISHED.equals(publication.getStatus()) && file.isVisibleForNonOwner();
    }
```
    
    I think it can be prettified :)